File: modules/java/main/src/main/java/com/mooltiverse/oss/nyx/entities/GitRemoteConfiguration.java
/*
 * Copyright 2020 Mooltiverse
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.mooltiverse.oss.nyx.entities;

/**
 * This object models the fields used to configure the remote Git repository.
 */
public class GitRemoteConfiguration {
    /**
     * The authentication method.
     */
    private AuthenticationMethod authenticationMethod = null;

    /**
     * The remote user name.
     */
    private String user = null;

    /**
     * The remote password.
     */
    private String password = null;

    /**
     * The private key.
     */
    private String privateKey = null;

    /**
     * The passphrase for the private key.
     */
    private String passphrase = null;

    /**
     * Default constructor.
     */
    public GitRemoteConfiguration() {
        super();
    }

    /**
     * Standard constructor.
     * 
     * @param authenticationMethod the authentication method.
     * @param user the remote user name.
     * @param password the remote password.
     * @param privateKey the private key.
     * @param passphrase the passphrase for the private key.
     */
    public GitRemoteConfiguration(AuthenticationMethod authenticationMethod, String user, String password, String privateKey, String passphrase) {
        super();
        this.authenticationMethod = authenticationMethod;
        this.user = user;
        this.password = password;
        this.privateKey = privateKey;
        this.passphrase = passphrase;
    }

    /**
     * Returns the authentication method.
     * 
     * @return the authentication method.
     */
    public AuthenticationMethod getAuthenticationMethod() {
        return authenticationMethod;
    }

    /**
     * Sets the authentication method.
     * 
     * @param authenticationMethod the authentication method.
     */
    public void setAuthenticationMethod(AuthenticationMethod authenticationMethod) {
        this.authenticationMethod = authenticationMethod;
    }

    /**
     * Returns the remote user name.
     * 
     * @return the remote user name.
     */
    public String getUser() {
        return user;
    }

    /**
     * Sets the remote user name.
     * 
     * @param user the remote user name.
     */
    public void setUser(String user) {
        this.user = user;
    }

    /**
     * Returns the remote password.
     * 
     * @return the remote password.
     */
    public String getPassword() {
        return password;
    }

    /**
     * Sets the remote password.
     * 
     * @param password the remote password.
     */
    public void setPassword(String password) {
        this.password = password;
    }

    /**
     * Returns the private key.
     * 
     * @return the private key.
     */
    public String getPrivateKey() {
        return privateKey;
    }

    /**
     * Sets the private key.
     * 
     * @param privateKey the private key.
     */
    public void setPrivateKey(String privateKey) {
        this.privateKey = privateKey;
    }

    /**
     * Returns the passphrase for the private key.
     * 
     * @return the passphrase for the private key.
     */
    public String getPassphrase() {
        return passphrase;
    }

    /**
     * Sets the passphrase for the private key.
     * 
     * @param passphrase the passphrase for the private key.
     */
    public void setPassphrase(String passphrase) {
        this.passphrase = passphrase;
    }
}

File: src/go/nyx/entities/git_remote_configuration.go
/*
 * Copyright 2020 Mooltiverse
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package entities

/*
This object models the fields used to configure the remote Git repository.

This structure is JSON and YAML aware so all objects are properly managed for marshalling and unmarshalling. This comes with a downside
as all internal fields must be exported (have the first capital letter in their names) or they can't be marshalled.
*/
type GitRemoteConfiguration struct {
	// The authentication method.
	AuthenticationMethod *AuthenticationMethod `json:"authenticationMethod,omitempty" yaml:"authenticationMethod,omitempty"`

	// The remote user name.
	User *string `json:"user,omitempty" yaml:"user,omitempty"`

	// The remote password.
	Password *string `json:"password,omitempty" yaml:"password,omitempty"`

	// The private key.
	PrivateKey *string `json:"privateKey,omitempty" yaml:"privateKey,omitempty"`

	// The passphrase for the private key.
	Passphrase *string `json:"passphrase,omitempty" yaml:"passphrase,omitempty"`
}

/*
Default constructor
*/
func NewGitRemoteConfiguration() *GitRemoteConfiguration {
	return &GitRemoteConfiguration{}
}

/*
Standard constructor.

Arguments are as follows:

- authenticationMethod the authentication method.
- user the remote user name.
- password the remote password.
- privateKey the private key.
- passphrase the passphrase for the private key.
*/
func NewGitRemoteConfigurationWith(authenticationMethod *AuthenticationMethod, user *string, password *string, privateKey *string, passphrase *string) *GitRemoteConfiguration {
	grc := GitRemoteConfiguration{}

	grc.AuthenticationMethod = authenticationMethod
	grc.User = user
	grc.Password = password
	grc.PrivateKey = privateKey
	grc.Passphrase = passphrase

	return &grc
}

/*
Returns the authentication method.
*/
func (grc *GitRemoteConfiguration) GetAuthenticationMethod() *AuthenticationMethod {
	return grc.AuthenticationMethod
}

/*
Sets the authentication method.
*/
func (grc *GitRemoteConfiguration) SetAuthenticationMethod(authenticationMethod *AuthenticationMethod) {
	grc.AuthenticationMethod = authenticationMethod
}

/*
Returns the remote user name.
*/
func (grc *GitRemoteConfiguration) GetUser() *string {
	return grc.User
}

/*
Sets the remote user name.
*/
func (grc *GitRemoteConfiguration) SetUser(user *string) {
	grc.User = user
}

/*
Returns the remote password.
*/
func (grc *GitRemoteConfiguration) GetPassword() *string {
	return grc.Password
}

/*
Sets the remote password.
*/
func (grc *GitRemoteConfiguration) SetPassword(password *string) {
	grc.Password = password
}

/*
Returns the private key.
*/
func (grc *GitRemoteConfiguration) GetPrivateKey() *string {
	return grc.PrivateKey
}

/*
Sets the private key.
*/
func (grc *GitRemoteConfiguration) SetPrivateKey(privateKey *string) {
	grc.PrivateKey = privateKey
}

/*
Returns the passphrase for the private key.
*/
func (grc *GitRemoteConfiguration) GetPassphrase() *string {
	return grc.Passphrase
}

/*
Sets the passphrase for the private key.
*/
func (grc *GitRemoteConfiguration) SetPassphrase(passphrase *string) {
	grc.Passphrase = passphrase
}
